cmd/appliance/logs.go

```diff
@@ -0,0 +1,167 @@
+	"github.com/appgate/sdpctl/pkg/docs"
+	"github.com/appgate/sdpctl/pkg/factory"
+	"github.com/appgate/sdpctl/pkg/terminal"
+	"github.com/appgate/sdpctl/pkg/tui"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/vbauerster/mpb/v7"
+	"github.com/vbauerster/mpb/v7/decor"
+)
+
+type logOpts struct {
+	cmdappliance.AppliancCmdOpts
+	Out        io.Writer
+	BaseURL    string
+	HTTPClient func() (*http.Client, error)
+	SpinnerOut func() io.Writer
+	Version    int
+	Path       string
+}
+
+func NewLogsCmd(f *factory.Factory) *cobra.Command {
+	aopts := cmdappliance.AppliancCmdOpts{
+		Appliance: f.Appliance,
+		Config:    f.Config,
+		CanPrompt: f.CanPrompt(),
+	}
+	opts := logOpts{
+		aopts,
+		f.IOOutWriter,
+		f.Config.URL,
+		f.CustomHTTPClient,
+		f.GetSpinnerOutput(),
+		f.Config.Version,
+		"",
+	}
+	cmd := &cobra.Command{
+		Use:     "logs",
+		Short:   docs.ApplianceLogsDoc.Short,
+		Long:    docs.ApplianceLogsDoc.Short,
+		Example: docs.ApplianceLogsDoc.ExampleString(),
+		Args: func(cmd *cobra.Command, args []string) error {
+			return cmdappliance.ArgsSelectAppliance(cmd, args, &opts.AppliancCmdOpts)
+		},
+		RunE: func(c *cobra.Command, args []string) error {
+			return logsRun(c, args, &opts)
+		},
+	}
+	cmd.Flags().StringVarP(&opts.Path, "path", "", "", "Optional path to write to")
+	return cmd
+}
+
+func logsRun(cmd *cobra.Command, args []string, opts *logOpts) error {
+	terminal.Lock()
+	defer terminal.Unlock()
+	client, err := opts.HTTPClient()
+	if err != nil {
+		return fmt.Errorf("could not resolve a HTTP client based on your current config %s", err)
+	}
+	path, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	if len(opts.Path) > 0 {
+		path = opts.Path
+	}
+
+	requestURL := fmt.Sprintf("%s/appliances/%s/logs", opts.BaseURL, opts.ApplianceID)
+	request, err := http.NewRequest(http.MethodGet, requestURL, nil)
+	if err != nil {
+		return err
+	}
+	request = request.WithContext(context.WithValue(context.Background(), factory.ContextAcceptValue, fmt.Sprintf("application/vnd.appgate.peer-v%d+zip", opts.Version)))
+	log.Infof("starting downloading log zip bundle or %s", opts.ApplianceID)
+	response, err := client.Do(request)
+	if response == nil || err != nil {
+		return api.HTTPErrorResponse(response, err)
+	}
+	if response.StatusCode != http.StatusOK {
+		return api.HTTPErrorResponse(response, err)
+	}
+	defer response.Body.Close()
+
+	name := fmt.Sprintf("%s_logs.zip", opts.ApplianceID)
+	_, params, err := mime.ParseMediaType(response.Header.Get("Content-Disposition"))
+	if err == nil {
+		if v, ok := params["filename"]; ok {
+			name = v
+		}
+	}
+
+	file, err := os.Create(filepath.Join(path, name))
+	if err != nil {
+		return err
+	}
+
+	p := mpb.New(mpb.WithWidth(64), mpb.WithOutput(opts.SpinnerOut()))
+
+	bar := p.New(0,
+		mpb.SpinnerStyle(tui.SpinnerStyle...),
+		mpb.BarFillerOnComplete(tui.Check),
+		mpb.BarWidth(1),
+		mpb.AppendDecorators(
+			decor.Name(name+" ", decor.WCSyncWidthR),
```

Comment:
The `decor.WCSyncWidthR` option here will not do anything since it only applies when you use multiple bars and you can only download from one appliance at a time.